packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -88,21 +88,24 @@ export class TdsDropdown {
   /** Method for setting the value of the Dropdown. */
   @Method()
   async setValue(newValue: string, newValueLabel: string) {
-    if (this.multiselect) {
-      this.selection = this.selection
-        ? [...this.selection, { value: newValue, label: newValueLabel }]
-        : [{ value: newValue, label: newValueLabel }];
-    } else {
-      this.selection = [{ value: newValue, label: newValueLabel }];
-      this.children = this.getChildren().map((element: HTMLTdsDropdownOptionElement) => {
-        if (element.value !== newValue) {
-          element.setSelected(false);
-        }
-        return element;
-      });
+    // Check if any of the dropdown options has the value that is passed to the method.
+    if (
```

Comment:
Maybe rewrite like this a little, for more readability?
```suggestion
    const optionExists = this.getChildren().some(
    (element: HTMLTdsDropdownOptionElement) => element.value === newValue
  );

  if (optionExists) {
    // Create a new selection based on the multiselect flag.
    const newSelection = this.multiselect
      ? [...(this.selection || []), { value: newValue, label: newValueLabel }]
      : [{ value: newValue, label: newValueLabel }];

    // Update the selection and the 'value' attribute of the host element.
    this.selection = newSelection;
    this.host.setAttribute('value', newSelection.map(selection => selection.value).toString());

    // Update the selected state of child options and trigger the change event.
    this.selectChildrenAsSelectedBasedOnSelectionProp();
    this.handleChange();
  }

  return this.selection;
```